garrysmod/lua/includes/modules/constraint.lua

```diff
@@ -340,30 +340,31 @@ function CreateKeyframeRope( Pos, width, material, Constraint, Ent1, LPos1, Bone
 
 	-- Attachment point 1
 	rope:SetEntity( "StartEntity", Ent1 )
-	rope:SetKeyValue( "StartOffset", tostring( LPos1 ) )
+	rope:SetKeyValue( "StartOffset", LPos1:ToString() )
 	rope:SetKeyValue( "StartBone", Bone1 )
-	
+
 	-- Attachment point 2
 	rope:SetEntity( "EndEntity", Ent2 )
-	rope:SetKeyValue( "EndOffset", tostring( LPos2 ) )
+	rope:SetKeyValue( "EndOffset", LPos2:ToString() )
 	rope:SetKeyValue( "EndBone", Bone2 )
-	
+
 	if ( kv ) then
 		for k, v in pairs( kv ) do
-		
-			rope:SetKeyValue( k, tostring( v ) )
-	
+
+			local value = ( isangle( v ) || isvector( v ) || ( istable( v ) && v.ToString ) ) && v:ToString() || tostring( v )

```

Comment:
You're passing keyvalues to the engine which accepts vectors etc as `X Y Z`, not `Angle(X, Y, Z)`.
Everything in this file should be using native tostring.